core/block_pkg_test.go

```diff
@@ -0,0 +1,61 @@
+package core
+
+import (
+	"testing"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTransactionCommitmentPoseidon(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		c, err := transactionCommitmentPoseidon(nil)
+		require.NoError(t, err)
+		assert.Equal(t, &felt.Zero, c)
+	})
+	t.Run("txs with signature", func(t *testing.T) {
+		var txs []Transaction
+
+		type signature = []*felt.Felt
+		// nil signature, empty signature and signature with some non-empty value
+		for i, sign := range []signature{nil, make(signature, 0), {new(felt.Felt).SetUint64(uint64(3))}} {
+			invoke := &InvokeTransaction{
+				TransactionHash:      new(felt.Felt).SetUint64(uint64(0 + i)),
+				TransactionSignature: sign,
+			}
+			deployAccount := &DeployAccountTransaction{
+				DeployTransaction: DeployTransaction{
+					TransactionHash: new(felt.Felt).SetUint64(uint64(1 + i)),
+				},
+				TransactionSignature: sign,
+			}
+			declare := &DeclareTransaction{
+				TransactionHash:      new(felt.Felt).SetUint64(uint64(2 + i)),
+				TransactionSignature: sign,
```

Comment:
why do we need 0+1, 1+i, 2+i (txn hashes still overlap with this right? We will have hashes 0,1,2,1,2,3,2,3,4)?